pkg/planner/core/indexmerge_path.go

```diff
@@ -596,7 +867,144 @@ func (ds *DataSource) generateIndexMergeOnDNF4MVIndex(normalPathCnt int, filters
+			Selection(remained-non-index-predicates)  --- COP
+				TableRowIdScan(t)  					  --- COP
+	2. select * from t where ((1 member of (a) and c=1) and (json_contains(b, '[1, 2, 3]') and d=2) and (other index predicates))
+		flatten as: select * from t where 1 member of (a) and json_contains(b, '[1, 2, 3]') and c=1 and c=2 and other index predicates
+		analyze: find and utilize index access filter items as much as possible:
+		IndexMerge(AND-INTERSECTION)                  --- ROOT
+			IndexRangeScan(mv-index-a)(1)             --- COP
+			IndexMerge(mv-index-b AND-INTERSECTION)   --- ROOT (embedded index merge) ---> simplify: we can defer the TableRowIdScan(t) to the outer index merge.
+				IndexRangeScan(a, [1,1])              --- COP
+				IndexRangeScan(a, [2,2])              --- COP
+				IndexRangeScan(a, [3,3])              --- COP
+			IndexRangeScan(non-mv-index-if-any)(?)    --- COP
+			TableRowIdScan(t)                         --- COP
+	3. select * from t where ((1 member of (a) and c=1) and (json_overlap(a, '[1, 2, 3]') and d=2) and (other index predicates))
+		flatten as: select * from t where 1 member of (a) and json_overlap(a, '[1, 2, 3]') and c=1 and c=2 and other index predicates
+		analyze: find and utilize index access filter items as much as possible:
+		IndexMerge(AND-INTERSECTION)                  --- ROOT
+			IndexRangeScan(mv-index-a)(1)             --- COP
+			IndexMerge(mv-index-b OR-UNION)           --- ROOT (embedded index merge) ---> simplify: we can defer the TableRowIdScan(t) to the outer index merge.
+				IndexRangeScan(a, [1,1])              --- COP
+				IndexRangeScan(a, [2,2])              --- COP
+				IndexRangeScan(a, [3,3])              --- COP
+			IndexRangeScan(non-mv-index-if-any)(?)    --- COP
+			TableRowIdScan(t)                         --- COP
+*/
+func (ds *DataSource) generateIndexMerge4ComposedIndex(normalPathCnt int,
+	indexMergeConds []expression.Expression) error {
+	isPossibleIdxMerge := len(indexMergeConds) > 0 && // have corresponding access conditions, and
+		len(ds.possibleAccessPaths) > 1 // have multiple index paths
+	if !isPossibleIdxMerge {
+		return nil
+	}
+	if strings.HasPrefix(ds.SCtx().GetSessionVars().StmtCtx.OriginalSQL, "explain select /*+ use_index_merge(t2, "+
+		"idx2, idx) */ * from t2 where ( json_overlaps") {
+		fmt.Println(1)
+	}
+
+	if len(indexMergeConds) == 1 {
+		// DNF path.
+		sf, ok := indexMergeConds[0].(*expression.ScalarFunction)
+		if !ok || sf.FuncName.L != ast.LogicOr {
+			// targeting: cond1 or cond2 or cond3
+			return nil
+		}
+		dnfFilters := expression.FlattenDNFConditions(sf)
+		mvIndexPartialPaths, usedAccessMap, needSelection4MVIndex, err := ds.generateIndexMergePartialPath4Or(normalPathCnt, dnfFilters)
+		if err != nil {
+			return err
+		}
+		remainedDNFItems := make([]expression.Expression, 0, len(dnfFilters))
+		for i, b := range usedAccessMap {
+			if !b {
+				remainedDNFItems = append(remainedDNFItems, dnfFilters[i])
+			}
+		}
```

Comment:
Just suggest that put all filters in the table-side Selection if `needSelection4MVIndex == True` for safety.

Reply:
yes, changed, for now, usedAccessMap is only used to identify whether one dnf item is completely not used, if so, composed-index-merge should be stopped soon.